modules/thermal_hydraulics/test/tests/components/deprecated/tests

```diff
@@ -0,0 +1,64 @@
+[Tests]
+  issues = '#20383'
+  [free_boundary]
+    type = 'RunException'
+    input = 'free_boundary.i'
+    expect_err = "Deprecated component. Use FreeBoundary1Phase or FreeBoundary2Phase instead."
+    max_parallel = 1
+    ad_indexing_type = 'global'
+    design = 'FreeBoundary.md'
+    requirement = 'The system shall throw an error if the FreeBoundary component is used.'
+  []
+  [gate_valve]
+    type = 'RunException'
+    input = 'gate_valve.i'
+    expect_err = "Deprecated component. Use GateValve1Phase or GateValve2Phase instead."
+    max_parallel = 1
+    ad_indexing_type = 'global'
+    design = 'GateValve.md'
+    requirement = 'The system shall throw an error if the GateValve component is used.'
+  []
+  [solid_wall]
+    type = 'RunException'
+    input = 'solid_wall.i'
+    expect_err = "Deprecated component. Use SolidWall1Phase or SolidWall2Phase instead."
+    max_parallel = 1
+    ad_indexing_type = 'global'
+    design = 'SolidWall.md'
+    requirement = 'The system shall throw an error if the SolidWall component is used.'
+  []
+  [junction_one_to_one]
+    type = 'RunException'
+    input = 'junction_one_to_one.i'
+    expect_err = "Depreacted component. Use JunctionOneToOne1Phase or JunctionOneToOne2Phase instead."
```

Comment:
"Depreacted" -> "Deprecated..."

Reply:
And, we will need to patch the error message in JunctionOneToOne.

Reply:
Good catch...